OpenRA.Mods.Common/ServerTraits/LobbySettingsNotification.cs

```diff
@@ -35,7 +35,7 @@ public void ClientJoined(OpenRA.Server.Server server, Connection conn)
 			foreach (var kv in server.LobbyInfo.GlobalSettings.LobbyOptions)
 			{
 				Session.LobbyOptionState def;
-				if (!defaults.LobbyOptions.TryGetValue(kv.Key, out def) || kv.Value.Value != def.Value)
+				if ((!defaults.LobbyOptions.TryGetValue(kv.Key, out def) || kv.Value.Value != def.Value) && options.ContainsKey(kv.Key))
```

Comment:
... although that's a bit of a mouthful, so it might be worth splitting out an inner `if` for readability if we go this way.